security/secret-protection/docs/secret-protection.rst

```diff
@@ -0,0 +1,281 @@
+`Security tutorial
+<https://docs.confluent.io/current/tutorials/security_tutorial.html>`__ shows
+you how to enable security features on the |cp|, but requires you to take extra
+steps in generating keys and certificates, and adding TLS configurations. In
+this tutorial instead, you'll encrypt a basic configuration parameter, which are
+the same steps.
+
+
+Generating the master encryption key based on a passphrase
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Select a master encryption key passphrase.
+
+   .. tip::
+
+        Your passphrase should be longer than the usual password. Choose a
+        phrase you can remember as a string of words.
+
+#. Enter the passphrase into a file to be passed into the CLI, to avoid logging
+   history showing the passphrase.
+
+#. Choose the location where the secrets file will reside on your local host.
+
+   .. note::
+
+       The location shouldn't be where the |cp| services run.
+
+   The secrets file will contain encrypted secrets for the master encryption
+   key, data encryption key, and configuration parameters, along with their
+   metadata, such as which cipher was used for encryption.
+
+#. Generate the master encryption key by running the following command:
+
+   .. code-block:: text
+
+      # passphrase: /path/to/passphrase.txt
+      # local-secrets-file: /path/to/secrets.txt
+      confluent secret master-key generate --local-secrets-file /path/to/secrets.txt --passphrase @/path/to/passphrase.txt
+
+   You should see:
+
+   .. code-block:: text
+
+      Save the master key. It cannot be retrieved later.
+      +------------+----------------------------------------------+
+      | Master Key | Nf1IL2bmqRdEz2DO//gX2C+4PjF5j8hGXYSu9Na9bao= |
+      +------------+----------------------------------------------+
+
+#. Save the master key somewhere safe on your computer.
+
+#. Export the key into the environment on the local host and every host
+   that will have a configuration file with secret protection by running the
+   following command:
+
+   .. code-block:: text
+
+      export CONFLUENT_SECURITY_MASTER_KEY=Nf1IL2bmqRdEz2DO//gX2C+4PjF5j8hGXYSu9Na9bao=
+
+   To protect the previous environment variable in a production host, you can set
+   the master encryption key at the process level instead of the global machine
+   level. For example, you could set it in the ``systemd`` overrides for executed
+   processes, restricting the environment directives file to root-only access.
+
+
+Encrypting the value of a configuration parameter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use a configuration parameter available in the configuration file example that
+ships with |cp|.
+
+To encrypt the parameter ``config.storage.topic`` in
+``$CONFLUENT_HOME/etc/schema-registry/connect-avro-distributed.properties``,
+complete the following steps:
+
+#. Make a backup of this file, because the CLI currently does in-place
```

Comment:
```suggestion
#. Make a backup of the file ``$CONFLUENT_HOME/etc/schema-registry/connect-avro-distributed.properties``, because the CLI currently does in-place
```